apollo-router-core/src/spec/fragments.rs

```diff
@@ -36,15 +36,15 @@ impl Fragments {
                         .selection_set()
                         .expect("the node SelectionSet is not optional in the spec; qed")
                         .selections()
-                        .map(|selection| {
+                        .filter_map(|selection| {
                             Selection::from_ast(
                                 selection,
                                 &FieldType::Named(type_condition.clone()),
                                 schema,
                                 0,
                             )
                         })
-                        .collect::<Option<_>>()?;
```

Comment:
this behaviour is different:
- before, if any selection returned by `from_ast` was None, it would refuse the entire list
- now, if one of the selections returns `None`, it is ignored

Reply:
Let's add a test that asserts this behavior, so we can revisit it if it doesnt match what we re trying to accomplish here

Reply:
Problem solved, it's now using `Result` to be able to really return an error instead of using an `Option` as a `Result`